nmap_dns.cc

```diff
@@ -793,11 +783,7 @@ static void read_evt_handler(nsock_pool nsp, nsock_event evt, void *) {
               }
               output_summary();
               stat_ok++;
-              /*Since we have got the answer we can avoid iterating over all answers*/
-              return;
             }
-          /*This is the case where we aren't able to parse the PTR record successfully*/
-          break;
```

Comment:
Leave this break! Otherwise we fall through to the next case without checking it. (e.g. this answer gets processed first as a PTR, then as a CNAME.